### PR TITLE
Disable blackfire PHP extension by default

### PR DESCRIPTION
**Pull request by IgorVitol**
### Description

At the moment **blackfire** PHP extension can not be switched off, since it is enabled not by **PHP_EXTENSIONS** ENV variable. You can check it by executing command below:

- `docker run --rm magento/magento-cloud-docker-php:7.4-cli-1.2.0 php -v`

Note the last message:

> with blackfire v1.43.0~linux-x64-non_zts74, https://blackfire.io, by Blackfire

Even if monitoring is not configured - it affects performance. Just compare load time for listing/product page with enabled/disabled blackfire extension after execution of "**setup:upgrade**" / "**cache:flush**" commands.

### Manual testing scenarios
1. Blackfire should not be enabled by default. Execute and verify output (here should be no any mentions about blackfire).
`docker run --rm magento/magento-cloud-docker-php:7.4-cli-1.2.0 php -v`

2. If you want to enable it - just add "**blackfire**" to the list of extensions via **PHP_EXTENSIONS** env variable:
`docker run --rm -e "PHP_EXTENSIONS=blackfire" magento/magento-cloud-docker-php:7.4-cli-1.2.0 php -v`

Inspect php version output. Now you should see that blackfire is enabled. Here should be no warning like that:

> warning: blackfire (blackfire) is already loaded!

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [x] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by aepod** [comment]
Please provide metrics for the purposed issue with performance. Blackfire is enabled in Cloud by default, and it has been enabled in every project I have had on cloud docker (lots of projects)  and has never seen performance issues. Nor does a simple google search of blackfire agent causes performance issues provide any results. 

I would prefer it if it was done as an option in the ece-docker build:compose, such as --without-blackfire.  This way it can remain the default, where it should be since it is used in validation on CI/CD environments etc. It would not be the worst thing if it was off by default and we provided a --with-blackfire option.  

Did you test blackfire after you removed, and readded it using your method?

However, either way, requiring custom docker commands is frowned upon, you will have to implement a little more logic to make it much easier for people and scripting to handle the toggle of this.

**Comment by IgorVitol** [comment]
@aepod 
>Blackfire is enabled in Cloud by default

That's not correct. By default it is disabled and you have to enable it manually.
- Default values: https://github.com/magento/magento-cloud/blob/master/.magento.app.yaml#L14
- Documentation: https://blackfire.io/docs/integrations/paas/magentocloud#configuration

>you will have to implement a little more logic to make it much easier for people and scripting to handle the toggle of this

The logic is here... no need in something else. The correct way on how to enable needed php extensions in cloud local env - add it to **PHP_EXTENSIONS** env variable:
- Entrypoint: https://github.com/magento/magento-cloud-docker/blob/develop/images/php/7.4-cli/docker-entrypoint.sh#L62
- Default extensions: https://github.com/magento/magento-cloud-docker/blob/develop/images/php/7.4-cli/Dockerfile#L14

By some reason blackfire implementation didn't follow that rule... so it is not possible to disable it at all.

**Review by arhiopterecs — CHANGES_REQUESTED** [review]
Hi @IgorVitol,
Thanks for your PR
Yes, that's a good idea
But now I agree with @aepod.
To add the ability to manage this extension now, we can add an option only "--without-blackfire"
We cannot change any default values in patch versions. Because it is BIC

Please add an option to disable this extension.

**Comment by IgorVitol** [comment]
@arhiopterecs Hi! Ok, thanks for reply.

Just to clarify - are we going to support existing/same docker image tags (e.g. 1.1, 1.2.0)  or here would be new docker image version tag for new "patch" build?

**Comment by oshmyheliuk** [comment]
@IgorVitol for each new clod-docker version will be added new images to avoid issues when the codebase is not compatible with images. 
So the next version of `magento-cloud-docker` will be 1.2.1 and cloud docker images will be `xxx-1.2.1`

**Comment by IgorVitol** [comment]
@arhiopterecs Pull request updated. Have enabled blackfire by default using PHP_EXTENSIONS env variable (in order to not change current behavior). Also added option "--without-blackfire" to remove it from default extensions (in PHP_EXTENSIONS env variable).

At least, it is possible now to disable it when not needed. Pls check if that's what you mean

File: src/Config/Source/CliSource.php [hunk 1]
```diff
@@ -234,6 +235,12 @@ public function read(): Repository
             ]);
         }
 
+        if ($this->input->getOption(self::OPTION_WITHOUT_BLACKFIRE)) {
+            $repository->set([
+                self::PHP_DISABLED_EXTENSIONS => ['blackfire']
+            ]);
```
Comment (arhiopterecs) [inline src/Config/Source/CliSource.php]: @IgorVitol 
It looks like you are overwriting the value but you need to add
https://github.com/magento/magento-cloud-docker/blob/1ee887057e4fa5f5a6c30c8e204dbd1bdcd39463/src/Config/Source/CloudSource.php#L244

```suggestion
            $repository->push(self::PHP_DISABLED_EXTENSIONS, 'blackfire');
```

**Review by arhiopterecs — CHANGES_REQUESTED** [review]

**Comment by IgorVitol** [comment]
@arhiopterecs It looks like [current implementation](https://github.com/magento/magento-cloud-docker/blob/develop/src/Config/Source/CloudSource.php#L225) takes into account enabled runtime extensions from cloud configuration file:
https://github.com/magento/magento-cloud/blob/master/.magento.app.yaml#L13

I just want to double check that we really need to add "**--without-blackfire**" option? Since if it was enabled here in customer repository, then it will still remain enabled and no any additional actions will be required for existing customers.

**Comment by IgorVitol** [comment]
@arhiopterecs Pull request updated. Looks like all sources are not merging, but [replaced recursively](https://github.com/magento/magento-cloud-docker/blob/develop/src/Config/Config.php#L67). So, we can not just "push" additional values - in our case we need to push it with specific array key, to make sure that it will not replace anything. This time I have added value with a key. Pls check once again.

Checked together with disabled extensions config - both values works in this case.

---
 @oshmyheliuk @arhiopterecs Also here is one more thing which confusing me - what is the correct flow to enable blackfire for local env now? Blackfire PHP extension in develop branch is always enabled on php containers, that's clear. But it will not properly work without enabled "blackfire" service & provided configuration for it...

I found next code, which should cover this:
https://github.com/magento/magento-cloud-docker/blob/develop/src/Config/Source/CloudSource.php#L223

But it assumes that blackfire should be enabled in some non-documented way in cloud [configuration file](https://github.com/magento/magento-cloud/blob/master/.magento.app.yaml#L13) (e.g. provided not as string value - "blackfire", but as an array with configuration). How this works then when such config file will be pushed to the cloud env?

Another possible solution - enable it as completely custom custom compose config...

In both cases "blackfire" php extension will be added to the **PHP_EXTENSIONS** env variable. So, it is confusing me once again - why we can not just disable extension by default in container and let it work when provided in the list? It will be in **PHP_EXTENSIONS** list anyway for any existing or new customers/developers.

**Review by arhiopterecs — CHANGES_REQUESTED** [review]
@IgorVitol 
Thank you for your patience
After discussing this issue with the team members and the product owner, it was decided to fix this defect in the patch version.
AC:
 - 'blackfire' is disabled by default
 - 'blackfire' is enabled by adding the extension to the file '.magento.app.yaml'

**Comment by oshmyheliuk** [comment]
@IgorVitol Hi,
Could you please resolve conflicts, and make the necessary changes as it was discussed with @arhiopterecs.
We are planning to make a release soon and would be nice to get this PR into release.

**Comment by NadiyaS** [comment]
@magento import pr to magento-commerce/magento-cloud-docker

**Comment by magento-engcom-team** [comment]
@NadiyaS the pull request successfully imported.

**Comment by NadiyaS** [comment]
Hi @IgorVitol ,
Since we are preparing the code for release,  we've already imported your PR to our internal repo to proceed with the delivery process. All conflicts will also be resolved from our side.
Thank you for your contribution.

**Comment by IgorVitol** [comment]
@NadiyaS This PR doesn't meet agreed AC. I have planned to do changes in the morning, but could do it in next 30 min if that's still fine for you

**Comment by NadiyaS** [comment]
@IgorVitol 
we are limited in time, but if you need just few hours, you still may push your changes and I will import them. But do not worry if you cannot, we will fix what was left.

**Comment by IgorVitol** [comment]
@NadiyaS @oshmyheliuk @arhiopterecs PR updated according to suggested AC:
>'blackfire' is disabled by default
>'blackfire' is enabled by adding the extension to the file '.magento.app.yaml'

**Review by arhiopterecs — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by NadiyaS** [comment]
@magento import pr to magento-commerce/magento-cloud-docker

**Comment by magento-engcom-team** [comment]
@NadiyaS, an error occurred during the Pull Request import.

**Comment by NadiyaS** [comment]
> @NadiyaS, an error occurred during the Pull Request import.

Please ignore this message. All updates were successfully imported.

**Pull request merged**